src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,109 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ * haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * 
+ * @class
+ * @param {string} name - the name of the client property
+ * @param {any} [val] - the initial value
+ * @param {function} [makeValid] - returns a valid value no
+ * matter what. Will be given the new value, the current value,
+ * the string 'current' or 'future', and the `FormValue` instance.
+ * 
+ * @returns FormValue
+ * @todo Finish description of FormValue return value
+ */
+class FormValue {
+
+  constructor ( name, val = null, makeValid ) {
+
+    var fVal = this;
+    
+    fVal.name = name;
+    fVal._futureWasChanged = false;  // Handled internally
```

Comment:
It's something I'm thinking about. Again, non-explicit getters and setters, and code in general, are things I tend to avoid. The getter solution was very elegant, though. Maybe I need to see it in action with a few of our other, more sticky, cases to see how well it handles there.

Reply:
I accept that the implicit behavior of propagation could raise a risk of black-boxing the behavior. I think that will be the case with either solution. If you'd strongly prefer the explicit state, I can live with it.